smartsim/settings/dispatch.py

```diff
@@ -378,10 +382,16 @@ class ExecutableProtocol(t.Protocol):
     def as_program_arguments(self) -> t.Sequence[str]: ...
 
 
-class LauncherProtocol(t.Protocol[_T_contra]):
-    def start(self, launchable: _T_contra, /) -> LaunchedJobID: ...
+class LauncherProtocol(collections.abc.Hashable, t.Protocol[_T_contra]):
     @classmethod
+    @abc.abstractmethod
     def create(cls, exp: Experiment, /) -> Self: ...
+    @abc.abstractmethod
+    def start(self, launchable: _T_contra, /) -> LaunchedJobID: ...
+    @abc.abstractmethod
+    def get_status(
+        self, *launched_ids: LaunchedJobID
+    ) -> tuple[SmartSimStatus, ...]: ...
```

Comment:
Currently, this assumes with this call:
```py
stat_1, stat_2, stat_3, ... = launcher.get_status(id_1, id_2, id_3, ...)
```
`stat_1` is the status of the job with id `id_1`, `stat_2` is the status of the job with id `id_2`, etc. and is behavior that is relied upon by the `Experiment`.

Do we like this or is this too much of an "implied constraint" from the protocol?

Should the return type here actually be a `Mapping[LaunchJobID, SmartSimStatus]` or maybe a `Collection[tuple[LaunchJobID, SmartSimStatus]]`? That way its more obvious to users looking to write and register their own launchers the intention of what this method should return.

Reply:
I feel like it might not be implied for all SS users that stat_1 and stat_2 will map to id_1 and id_2 but I might be wrong. I would rather get back a Mapping or Iterable -> Im wondering what will be easier for the user to inspect, maybe an iterable since you can use a for loop? But then again I can search via key in a Mapping - am I on to something here or is this so far off?

Reply:
Agreed offline to make this protocol method return a `Mapping`.